snakemake_storage_plugin_xrootd/__init__.py

```
@@ -129,9 +131,26 @@ def __post_init__(self):
+                return getattr(module, func_name)
+            except (ImportError, AttributeError):
+                # Fallback to eval if loading as entry point fails
+                return None
+        return None
 
     def url_decorator(self, url: str) -> str:
+        if self.dec_func is not None:
+            return self.dec_func(url)
         if self.settings.url_decorator is not None:
+            # Fallback for backwards compatibility (eval)
             return eval(self.settings.url_decorator, {"url": url})
```

Ruff (0.14.14)

[error] Use of possibly insecure function; consider using `ast.literal_eval`

(S307)